Adding Polymarket Cookbook (#858)
graphprotocol/docs: website/pages/en/subgraphs/cookbook/polymarket.mdx

---
title: Querying Blockchain Data from Polymarket with Subgraphs on The Graph
---

> Polymarket’s onchain data can be easily queried using GraphQL via subgraphs on The Graph Network. Subgraphs are decentralized APIs powered by The Graph, a protocol for indexing & querying data from blockchains.

Reviewer: ```suggestion
Query Polymarket’s onchain data using GraphQL via subgraphs on The Graph Network. Subgraphs are decentralized APIs powered by The Graph, a protocol for indexing & querying data from blockchains.
```

> ## The Polymarket Subgraph

Reviewer: ```suggestion
## Polymarket Subgraph on Graph Explorer 
```

You can see an interactive query playground on the [Polymarket subgraph’s page on The Graph Explorer](https://thegraph.com/explorer/subgraphs/Bx1W4S7kDVxs9gC3s2G6DS8kdNBJNVhMviCtin2DiBp?view=Query&chain=arbitrum-one), where you can test any query.

![Polymarket Playground](/img/Polymarket-playground.png)

> ## How to use the Visual Query Editor

Reviewer: - It might be useful to share, why they might want to use this

        "payout": "2135448291991",
        "redeemer": "0x5a181dcf3eb53a09fb32b20a5a9312fb8d26f689",
        "timestamp": "1704932625"
      },
      {
        "id": "0x2b2826448fcacde7931828cfcd3cc4aaeac8080fdff1e91363f0589c9b503eca_0xa",
        "payout": "1917395333835",
        "redeemer": "0xfffe4013adfe325c6e02d36dc66e091f5476f52c",
        "timestamp": "1726701528"
      },
      {
        "id": "0xfe82e117201f5169abc822281ccf0469e6b3740fcb4e799d1b599f83b8f11656_0x30",
        "payout": "1862505580000",
        "redeemer": "0xfffe4013adfe325c6e02d36dc66e091f5476f52c",
        "timestamp": "1722929866"
      }
    ]
  }
}
```

## Polymarket's GraphQL Schema

The schema for this subgraph is defined [here in Polymarket’s GitHub](https://github.com/Polymarket/polymarket-subgraph/blob/main/polymarket-subgraph/schema.graphql).

### Polymarket Subgraph Endpoint

https://gateway.thegraph.com/api/{api-key}/subgraphs/id/Bx1W4S7kDVxs9gC3s2G6DS8kdNBJNVhMviCtin2DiBp

The Polymarket Subgraph endpoint is available on [Graph Explorer](https://thegraph.com/explorer).

![Polymarket Endpoint](/img/Polymarket-endpoint.png)

> ## How to FGet your own API Key

Reviewer: ```suggestion
## How to Get your own API Key
```

- [Polymarket Profit & Loss](https://thegraph.com/explorer/subgraphs/6c58N5U4MtQE2Y8njfVrrAfRykzfqajMGeTMEvMmskVz?view=Query&chain=arbitrum-one)
- [Polymarket Open Interest](https://thegraph.com/explorer/subgraphs/ELaW6RtkbmYNmMMU6hEPsghG9Ko3EXSmiRkH855M4qfF?view=Query&chain=arbitrum-one)

## How to Query with the API

You can pass any GraphQL query to the Polymarket endpoint and receive data in json format.

This following code example will return the exact same output as above.

### Sample Code from node.js

```
const axios = require('axios');

const graphqlQuery = `{
  positions(first: 5) {
    condition
    outcomeIndex
  }
};

const queryUrl = 'https://gateway.thegraph.com/api/{api-key}/subgraphs/id/Bx1W4S7kDVxs9gC3s2G6DS8kdNBJNVhMviCtin2DiBp'

const graphQLRequest = {
  method: 'post',
  url: queryUrl,
  data: {
    query: graphqlQuery,
  },
};

// Send the GraphQL query
axios(graphQLRequest)
  .then((response) => {
    // Handle the response here
    const data = response.data.data
    console.log(data)

  })
  .catch((error) => {
    // Handle any errors
    console.error(error);
  });
```

> ## About The Graph

Reviewer: - Not sure why this is listed here
- I think this would be a terrific add in to a page that discusses querying in general
